packages/langium/src/parser/token-builder.ts

```diff
@@ -87,20 +79,19 @@ export class DefaultTokenBuilder implements TokenBuilder {
             keyword.value;
     }
 
-    protected findLongerAlt(keyword: Keyword, keywords: Keyword[], terminals: TerminalRule[], tokenMap: Map<string, TokenType>): TokenType[] {
+    protected findLongerAlt(keyword: Keyword, keywordTokens: TokenType[], terminalTokens: TokenType[]): TokenType[] {
```

Comment:
I've though about this for a bit, and came to the conclusion that we don't need the `keywordTokens` parameter here: We already sort keywords by descending length, therefore, no keyword is a longer alternative of another keyword. Shorter keywords can never be matched before a longer keyword anyway.

Removing this would simplify the `buildKeywordTokens` method as well.

Reply:
Unfortunately, we can't remove it from `keywordTokens`.

> Shorter keywords can never be matched before a longer keyword anyway.
Yes, it's true, but we handle keywords in descending order by their length, so we handle longer keywords before shorter ones. As the result, when we handle shorter keywords, they already can have longer alternatives in `keywordTokens`.

For example, keywords `?=` and `?`. `?=` is a longer alternative of `?` that is found only thanks to using `keywordTokens`.

Reply:
Yes, but Chevrotain will always try to lex `?=` before attempting `?`. That's why we don't need the `longerAlt` property on keyword tokens for other keywords. There is no scenario where Chevrotain attempts to match `?` even though the next two characters are `?=`.